apinetlet/controllers/network_peering_controller.go

```diff
@@ -0,0 +1,270 @@
+	WatchFilterValue string
+}
+
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+//+kubebuilder:rbac:groups=networking.ironcore.dev,resources=networks,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=networking.ironcore.dev,resources=networks/status,verbs=get;update;patch
+
+//+cluster=apinet:kubebuilder:rbac:groups=core.apinet.ironcore.dev,resources=networks,verbs=get;list;watch;create;update;patch;delete;deletecollection
+
+func (r *NetworkPeeringReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	network := &networkingv1alpha1.Network{}
+	if err := r.Get(ctx, req.NamespacedName, network); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	return r.reconcileExists(ctx, log, network)
+}
+
+func (r *NetworkPeeringReconciler) reconcileExists(ctx context.Context, log logr.Logger, network *networkingv1alpha1.Network) (ctrl.Result, error) {
+	if !network.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	return r.reconcile(ctx, log, network)
+}
+
+func (r *NetworkPeeringReconciler) reconcile(ctx context.Context, log logr.Logger, network *networkingv1alpha1.Network) (ctrl.Result, error) {
+	log.V(1).Info("Reconcile")
+
+	var peeringClaimRefs []networkingv1alpha1.NetworkPeeringClaimRef
+	var peeringNames []string
+	var peeredIDs []string
+
+	for _, peering := range network.Spec.Peerings {
+		peeringClaimRef, peeringName, peeredID, err := r.reconcilePeering(ctx, log, network, peering)
+
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("[network peering %s] %w", peering.Name, err)
+		}
+
+		if peeringClaimRef != (networkingv1alpha1.NetworkPeeringClaimRef{}) {
+			peeringClaimRefs = append(peeringClaimRefs, peeringClaimRef)
+		}
+		if peeringName != "" {
+			peeringNames = append(peeringNames, peeringName)
+		}
+		if peeredID != "" {
+			peeredIDs = append(peeredIDs, peeredID)
+		}
+	}
+
+	if len(peeringClaimRefs) > 0 {
+		log.V(1).Info("Peering claim refs require network spec update")
+		if err := r.updateSpec(ctx, log, network, peeringClaimRefs, peeredIDs); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error updating network spec: %w", err)
+		}
+	}
+
+	if len(peeringNames) > 0 {
+		log.V(1).Info("Network peering status require network status update")
+		if err := r.updateStatus(ctx, log, network, peeringNames); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error updating network status: %w", err)
+		}
+	}
+
+	log.V(1).Info("Reconciled")
+	return ctrl.Result{}, nil
+}
+
+func (r *NetworkPeeringReconciler) updateStatus(
+	ctx context.Context,
+	log logr.Logger,
+	network *networkingv1alpha1.Network,
+	peeringNames []string,
+) error {
+	base := network.DeepCopy()
+	newStatusPeerings := make([]networkingv1alpha1.NetworkPeeringStatus, 0, len(peeringNames))
+	for _, name := range peeringNames {
+		newStatusPeerings = append(newStatusPeerings, networkingv1alpha1.NetworkPeeringStatus{
+			Name: name,
```

Comment:
add a short TODO to indicate this needs to be extended once the PR in ironcore for status extension has been merged